### PR TITLE
feat(perf): add a search bar to the span details page

### PR DESCRIPTION
**Pull request by udameli**
This PR adds a search bar to the span details page. It reads query parameters from the url and populates the search bar with the values and, when updated, pushes new search query to the url. Aggregate search fields have no effect on the queries but they can be entered in the search bar. I will consult with Dora on a user-friendly approach to communicate to users that aggregate fields will be ignored in the queries.

Fixes PERF-1404

### CODE REVIEW

File: static/app/views/performance/transactionSummary/transactionSpans/spanDetails/spanDetailsSearchBar.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import styled from '@emotion/styled';
+import {Location} from 'history';
+
+import _SearchBar from 'sentry/components/events/searchBar';
```
Comment (evanpurkhiser) [inline static/app/views/performance/transactionSummary/transactionSpans/spanDetails/spanDetailsSearchBar.tsx:4]: I would avoid this and instead probably call it `StyledSearchBar = styled(SearchBar)`

Comment (udameli) [reply static/app/views/performance/transactionSummary/transactionSpans/spanDetails/spanDetailsSearchBar.tsx]: ah right, that takes us back to the default export vs named export argument
implemented your suggestion 👍

File: static/app/views/performance/transactionSummary/transactionSpans/spanDetails/spanDetailsHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
+import styled from '@emotion/styled';
+
+import {SectionHeading as _SectionHeading} from 'sentry/components/charts/styles';
```
Comment (evanpurkhiser) [inline static/app/views/performance/transactionSummary/transactionSpans/spanDetails/spanDetailsHeader.tsx:3]: same comment here for consistency 😌

File: static/app/views/performance/transactionSummary/transactionSpans/spanDetails/content.tsx [hunk 3]
```diff
@@ -283,51 +186,3 @@ function getSpansEventView(eventView: EventView): EventView {
   ];
   return eventView;
 }
-
-const ContentHeader = styled('div')`
```
Comment (udameli) [inline static/app/views/performance/transactionSummary/transactionSpans/spanDetails/content.tsx:287]: this component now has its own file which IMO simplified this file a bit

File: static/app/views/performance/transactionSummary/transactionSpans/spanDetails/spanDetailsSearchBar.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      query={query}
+      fields={eventView.fields}
+      onSearch={handleSearchQuery}
+    />
+  );
+}
+
+const StyledSearchBar = styled(SearchBar)`
+  margin-bottom: ${space(2)};
+`;
```
Comment (evanpurkhiser) [inline static/app/views/performance/transactionSummary/transactionSpans/spanDetails/spanDetailsSearchBar.tsx:50]: If you really wanna get rid of `StyledSearchBar` you could write it as

```suggestion
const SpanDetailsSearchBar = styled(({
  organization,
  eventView,
  location,
  className,
}: SpanDetailsSearchBarProps) => {
  const query = decodeScalar(location.query.query, '');

  const handleSearchQuery = (searchQuery: string): void => {
    browserHistory.push({
      pathname: location.pathname,
      query: {
        ...location.query,
        cursor: undefined,
        query: String(searchQuery).trim() || undefined,
      },
    });
  };

  return (
    <SearchBar
      className={className}
      placeholder={t('Filter Transactions')}
      organization={organization}
      projectIds={eventView.project}
      query={query}
      fields={eventView.fields}
      onSearch={handleSearchQuery}
    />
  );
})`
  margin-bottom: ${space(2)};
`;

export default SpanDetailsSearchBar
```

but I'm indifferent

File: static/app/views/performance/transactionSummary/transactionSpans/spanDetails/spanDetailsHeader.tsx (hunk 2 above)
Comment (udameli) [reply static/app/views/performance/transactionSummary/transactionSpans/spanDetails/spanDetailsHeader.tsx:3]: done 👍

File: static/app/views/performance/transactionSummary/transactionSpans/spanDetails/spanDetailsSearchBar.tsx (hunk 4 above)
Comment (udameli) [reply static/app/views/performance/transactionSummary/transactionSpans/spanDetails/spanDetailsSearchBar.tsx:50]: hm I'd prefer here to create a separate styled component to separate concerns a bit so component styles and logic are visibly separated

Comment (evanpurkhiser) [reply static/app/views/performance/transactionSummary/transactionSpans/spanDetails/spanDetailsSearchBar.tsx:50]: One functional difference here to recognize btw, is when you do what I suggested, you can continue to use it with `styled(SpanDetailsSearchBar)` or as a selector in other styled components. (Not that you would need to do that here, but worth being aware of!)

Comment (udameli) [reply static/app/views/performance/transactionSummary/transactionSpans/spanDetails/spanDetailsSearchBar.tsx:50]: Good to know 🤔

File: static/app/views/performance/transactionSummary/transactionSpans/spanDetails/spanDetailsHeader.tsx [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+  gap: ${space(3)};
+`;
+
+export const SpanLabelContainer = styled('div')`
+  ${overflowEllipsis};
+`;
+
+const EmptyValueContainer = styled('span')`
+  color: ${p => p.theme.gray300};
+`;
```
Comment (evanpurkhiser) [inline static/app/views/performance/transactionSummary/transactionSpans/spanDetails/spanDetailsHeader.tsx:152]: This stuff didn't really change right?

Comment (udameli) [reply static/app/views/performance/transactionSummary/transactionSpans/spanDetails/spanDetailsHeader.tsx:152]: nope, I just moved the header to its file

File: static/app/views/performance/transactionSummary/transactionSpans/spanDetails/spanDetailsSearchBar.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  const handleSearchQuery = (searchQuery: string): void => {
+    browserHistory.push({
+      pathname: location.pathname,
+      query: {
+        ...location.query,
+        cursor: undefined,
+        query: String(searchQuery).trim() || undefined,
+      },
+    });
+  };
```
Comment (k-fish) [inline static/app/views/performance/transactionSummary/transactionSpans/spanDetails/spanDetailsSearchBar.tsx:34]: Not for this PR, but we should probably just make this a util considering it's basically the same for all our search bars (+ eventView conditions + cursors).

**Review by k-fish — APPROVED** [review]
LGTM, comments aren't blocking.

File: static/app/views/performance/transactionSummary/transactionSpans/spanDetails/spanDetailsSearchBar.tsx (hunk 6 above)
Comment (udameli) [reply static/app/views/performance/transactionSummary/transactionSpans/spanDetails/spanDetailsSearchBar.tsx:34]: agreed 👍

File: tests/js/spec/views/performance/transactionSpans/spanDetails.spec.tsx [hunk 7]
```diff
@@ -300,6 +300,21 @@ describe('Performance > Transaction Spans > Span Summary', function () {
...
+        query: {project: '1', transaction: 'transaction'},
+      });
+
+      render(<SpanDetails params={{spanSlug: 'op:aaaaaaaa'}} {...data} />, {
+        context: data.routerContext,
+        organization: data.organization,
+      });
+
+      const searchBarNode = screen.queryByPlaceholderText('Filter Transactions');
+      expect(searchBarNode).not.toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (udameli) [reply tests/js/spec/views/performance/transactionSpans/spanDetails.spec.tsx:315]: yeah I see that for sure. I was gonna follow up with a PR that also tests browser history changes on search, I'll add a test for enabling/disabling histogram feature and checking whether the search bar is rendered

**Pull request merged**